### PR TITLE
API to compare sources from different drivers

### LINKED ISSUE #84: API to check whether two sources are equivalent
If we want to compare two data sources within a single driver, that is easy and we have API for it. But if we want to compare two data sources from two drivers, we do not currently support that. It would be useful to have it for mergin-db-sync (to compare e.g. a geopackage vs a postgres schema) - whether they have the same data structure and the same data inside.

It does not need to be super efficient, at least not in the initial implementation (for example, we could make a copy of data from one driver to format of the other driver and then use the "simple" comparison internally).

### PR DESCRIPTION
**Pull request by tomasMizera**
Introduced new API called `GEODIFF_createChangesetDr` to create changeset across two drivers. It works by first converting drivers to sqlite and then calling classic createChangeset. 

Possible enhancement: We do not need to call `makeCopy` API, just create schema from source driver and convert it to destination driver schema, then call createChangeset on dest driver or vice versa.

Usage:
``` cpp
GEODIFF_EXPORT int GEODIFF_createChangesetDr(
  const char *driverSrcName,
  const char *driverSrcExtraInfo,
  const char *src,
  const char *driverDstName,
  const char *driverDstExtraInfo,
  const char *dst,
  const char *changeset );
```

Resolves #84

### CODE REVIEW

**Comment by tomasMizera** [comment]
@wonder-sk added Python API

File: geodiff/src/geodiff.cpp [hunk 1]
```diff
@@ -106,6 +106,54 @@ int GEODIFF_createChangesetEx( const char *driverName, const char *driverExtraIn
 }
 
 
+int GEODIFF_createChangesetDr( const char *driverSrcName, const char *driverSrcExtraInfo, const char *src,
+                               const char *driverDstName, const char *driverDstExtraInfo, const char *dst,
+                               const char *changeset )
+{
+  if ( !driverSrcName || !driverDstName || !src || !dst || !changeset )
```
Comment (PeterPetrik) [inline geodiff/src/geodiff.cpp:113]: driver*ExtraInfo can be null?

File: geodiff/src/geodiff.cpp [hunk 2]
```diff
@@ -106,6 +106,54 @@ int GEODIFF_createChangesetEx( const char *driverName, const char *driverExtraIn
...
+    return GEODIFF_ERROR;
+  }
+
+  if ( strcmp( driverSrcName, driverDstName ) == 0 )
+  {
+    return GEODIFF_createChangesetEx( driverSrcName, driverSrcExtraInfo, src, dst, changeset );
+  }
+
+  // copy both sources to geopackage and create changeset
+  TmpFile tmpSrcGpkg( "" );
```
Comment (PeterPetrik) [inline geodiff/src/geodiff.cpp:125]: can we have constuctor with no arguments for TmpFile, so we do not need this ugly call?

File: pygeodiff/tests/test_single_commit.py [hunk 3]
```diff
@@ -45,7 +45,7 @@ def basetest(
   os.remove(changeset2)
   geodiff.invert_changeset(changeset, changeset_inv)
   geodiff.apply_changeset(patched, changeset_inv)
-  geodiff.create_changeset( patched, base, changeset2 )
+  geodiff.create_changeset_dr( "sqlite", "", patched, "sqlite", "", base, changeset2 )
```
Comment (PeterPetrik) [inline pygeodiff/tests/test_single_commit.py:48]: why this is changed?

**Review by PeterPetrik — COMMENTED** [review]
beautiful

File: pygeodiff/tests/test_single_commit.py (hunk 3 above)
Comment (tomasMizera) [reply pygeodiff/tests/test_single_commit.py:48]: Just to test if the new API is working from python

Comment (PeterPetrik) [reply pygeodiff/tests/test_single_commit.py:48]: maybe keep both?

Comment (tomasMizera) [reply pygeodiff/tests/test_single_commit.py:48]: Well, the create_changeset is called several times before and this should do the same thing. But I can add a separate test for new API

File: geodiff/src/geodiffutils.cpp [hunk 4]
```diff
@@ -506,6 +506,10 @@ std::string wstringToString( const std::wstring &wStr )
   }
 }
 
+TmpFile::TmpFile()
```
Comment (PeterPetrik) [inline geodiff/src/geodiffutils.cpp:509]: TmpFile::TmpFile() = default;

**Review by PeterPetrik — APPROVED** [review]

**Pull request merged**